dvc/cache/base.py

```diff
@@ -101,40 +123,53 @@ def changed(self, path_info, hash_info):
         Args:
             path_info: dict with path information.
             hash: expected hash value for this data.
+            filter_info: an optional argument to target a specific path.
 
         Returns:
             bool: True if data has changed, False otherwise.
         """
 
+        hash_origin = path_info
```

Comment:
This assignment is probably not needed. You can probably just use `path_info`.

Reply:
Not really, this is the origin of the `hast_info`. If we can successfully filter it, we change it below to the `filter_info`. And when comparing hashes, we use this location for obtaining the current cache